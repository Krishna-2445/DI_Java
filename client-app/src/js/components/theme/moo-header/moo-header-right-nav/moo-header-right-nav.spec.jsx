import React from 'react';
import { shallow } from 'enzyme';
import { MenuItem, Icon } from 'semantic-ui-react';
import MooHeaderRightNav from './moo-header-right-nav';

describe('<MooHeaderRightNav />', () => {
  const links = [
    { text: 'Rock', to: '/rock', icon: 'hand rock' },
    { text: 'Paper', to: '/paper', icon: 'hand paper' },
    { text: 'Scissors', to: '/scissors', icon: 'hand scissors' },
    { text: 'Lizard', to: '/lizard', icon: 'hand lizard' },
    { text: 'Spock', to: '/spock', icon: 'hand spock' },
  ];

  it('renders correct text, links, and icons', () => {
    const wrapper = shallow(<MooHeaderRightNav links={links} />);
    expect(wrapper.find(MenuItem)).toHaveLength(5);
    expect(wrapper.find(Icon)).toHaveLength(5);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders children', () => {
    const child = <div>Child</div>;
    const wrapper = shallow(<MooHeaderRightNav links={links}> {child} </MooHeaderRightNav>);
    expect(wrapper.contains('Child')).toEqual(true);
  });
});
