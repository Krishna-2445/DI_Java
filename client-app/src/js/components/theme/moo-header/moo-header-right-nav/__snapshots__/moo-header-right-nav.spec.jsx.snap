// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MooHeaderRightNav /> renders correct text, links, and icons 1`] = `
<Menu
  borderless={true}
  className="moo-right-nav"
  floated="right"
  stackable={true}
>
  <MenuItem
    activeClassName="active"
    as={[Function]}
    to="/rock"
  >
    <Icon
      as="i"
      name="hand rock"
    />
    Rock
  </MenuItem>
  <MenuItem
    activeClassName="active"
    as={[Function]}
    to="/paper"
  >
    <Icon
      as="i"
      name="hand paper"
    />
    Paper
  </MenuItem>
  <MenuItem
    activeClassName="active"
    as={[Function]}
    to="/scissors"
  >
    <Icon
      as="i"
      name="hand scissors"
    />
    Scissors
  </MenuItem>
  <MenuItem
    activeClassName="active"
    as={[Function]}
    to="/lizard"
  >
    <Icon
      as="i"
      name="hand lizard"
    />
    Lizard
  </MenuItem>
  <MenuItem
    activeClassName="active"
    as={[Function]}
    to="/spock"
  >
    <Icon
      as="i"
      name="hand spock"
    />
    Spock
  </MenuItem>
</Menu>
`;
